source/brailleDisplayDrivers/braillePen12.py

```diff
@@ -0,0 +1,182 @@
+			"braille_toggleTether": ("br(braillePen12):space+dot1+dot2+dot6",),
+			"kb:home": ("br(braillePen12):space+dot2+dot3",),
+			"kb:end": ("br(braillePen12):space+dot5+dot6",),
+			"review_top": ("br(braillePen12):space+dot1+dot2+dot3",),			
+			"review_bottom": ("br(braillePen12):space+dot4+dot5+dot6",),
+			"sayAll": ("br(braillePen12):space+dot1+dot2+dot4+dot5+dot6",),
+			"review_currentCharacter": ("br(braillePen12):space+dot3+dot6",),
+			"review_currentWord": ("br(braillePen12):space+dot2+dot5",),
+			"review_currentLine": ("br(braillePen12):space+dot1+dot4",),
+			"dateTime": ("br(braillePen12):space+dot2+dot3+dot4",),
+			"reportCurrentSelection": ("br(braillePen12):space+dot1+dot2+dot4+dot5+dot6+dot7",),
+			"showGui": ("br(braillePen12):space+dot1+dot2+dot4+dot5",),
+			"kb:alt+tab": ("br(braillePen12):space+dot1+dot3+dot5",),
+			"say_battery_status": ("br(braillePen12):space+dot1+dot2+dot7",),
+			"review_markStartForCopy": ("br(braillePen12):space+dot1+dot4+dot7",),
+			"review_copy": ("br(braillePen12):space+dot1+dot4+dot8",),
+			"reportClipboardText": ("br(braillePen12):space+dot1+dot2+dot3+dot6+dot8",),
+			"kb:control+v": ("br(braillePen12):space+dot1+dot2+dot3+dot6+dot7",),
+			"review_sayAll": ("br(braillePen12):space+dot1+dot3+dot4+dot5+dot6",),
+			"reviewMode_next": ("br(braillePen12):space+joyRight+dot7",),
+			"reviewMode_previous": ("br(braillePen12):space+joyLeft+dot7",),
+			"review_activate": ("br(braillePen12):joyAction+space",),
+			"title": ("br(braillePen12):space+dot2+dot3+dot4+dot5",),
+			"reportCurrentFocus": ("br(braillePen12):space+dot1+dot2+dot4+dot7",),
+			"reportStatusLine": ("br(braillePen12):space+dot2+dot3+dot4+dot7",),
+			"speakForeground": ("br(braillePen12):space+dot1+dot2+dot4+dot5+dot7",),
+			"toggleCaretMovesReviewCursor": ("br(braillePen12):joyAction+space+dot7",),
+			"kb:control+leftArrow": ("br(braillePen12):space+joyLeft+dot8",),
+			"kb:control+rightArrow": ("br(braillePen12):space+joyRight+dot8",),
+			"kb:control+home": ("br(braillePen12):space+joyUp+dot8",),
+			"kb:control+end": ("br(braillePen12):space+joyDown+dot8",),
+		},
+	})
+
+class InputGesture(braille.BrailleDisplayGesture, brailleInput.BrailleInputGesture):
+
+	source = BrailleDisplayDriver.name
+
+	def __init__(self, brailleKeys,controlKeys,routingKey):
+		super(InputGesture, self).__init__()	
+		brailleKeysList=[BP12_BRAILLE_KEYS[num] for num in xrange(8) if (brailleKeys>>num)&1]
+		controlKeysList=[BP12_CONTROL_KEYS[num] for num in xrange(8) if (controlKeys>>num)&1]
+		routingKeyList = ["routing"]
+		if routingKey==0:
+			selectedKeys=controlKeysList+brailleKeysList
+		else:
+			selectedKeys=controlKeysList+brailleKeysList+routingKeyList
+		self.id="+".join(set(selectedKeys))
+		self.keys = set(selectedKeys)
+		self.keyNames = names = set()
+		if brailleKeys!=0 and controlKeys==0 and routingKey==0:
+			self.dots =brailleKeys
+		if brailleKeys==0 and controlKeys==64 and routingKey==0:
+			self.space = True
```

Comment:
As with the Slim driver, this means `space` is only True if there are no dots. Is this intentional?